apps/examples/fs_test/smartfs_test.c

```diff
@@ -0,0 +1,337 @@
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/time.h>
+#include <fcntl.h>
+#include <errno.h>
+
+/****************************************************************************
+ * Definitions
+ ****************************************************************************/
+#define USAGE                                                           \
+	"\n FS R/W Performance Test\n"                                      \
+	"\n Usage :     smartfs_test <operation> <path> <size> <iteration> <print_option>\n"  \
+	"\nOperation :  1.Read/Write / 2.Read / 3.Write / 4. Write/Remove \n"                  \
+	" Path :        Test File Path\n"                                   \
+	" Size :        Write/Read Data size\n"                             \
+	" Iteration :   Number of repetitions\n"                            \
+	" Detail :      Enable(y) or Disable(n) Printing Details Option\n"  \
+	"               (Printing is affects at R/W Performance)\n"         \
+	"\n ex) smartfs_test 1 /mnt/test 4096 30 y\n"
+
+#define FS_RW_ITERATION_MAX 10000
+
+enum TEST_OPTS {
+	TEST_OPT_WRITEREAD = 1,
+	TEST_OPT_READ = 2,
+	TEST_OPT_WRITE = 3,
+	TEST_OPT_WRITEREMOVE = 4,
+	TEST_OPT_MAX,
+};
+
+/****************************************************************************
+ * Global Variables
+ ****************************************************************************/
+static int g_total_time;
+char g_printopt;
+
+/****************************************************************************
+ * smartfs_test
+ ****************************************************************************/
+
+static inline long get_time(void)
+{
+	struct timeval tv;
+
+	gettimeofday(&tv, NULL);
+	return (tv.tv_sec * 1000 + tv.tv_usec / 1000);
+}
+
+static void print_log(const char *format, ...)
+{
+	va_list ap;
+
+	if (g_printopt == 'y') {
+		va_start(ap, format);
+		vfprintf(stdout, format, ap);
+		va_end(ap);
+	}
+}
+
+static int fs_close(int fd)
+{
+	int ret;
+	long time;
+	long timediff;
+
+	time = get_time();
+	ret = close(fd);
+	if (ret < 0) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		return ERROR;
+	}
+	timediff = get_time() - time;
+	g_total_time += timediff;
+	print_log("close:%dms, ", timediff);
+
+	return OK;
+}
+
+static int fs_write(char *filepath, char *buf, size_t nbytes)
+{
+	int fd;
+	int ret;
+	int total_write = 0;
+	long time;
+	long timediff;
+
+	fd = open(filepath, O_WRONLY | O_CREAT | O_TRUNC);
+	if (fd < 0) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		return ERROR;
+	}
+
+	time = get_time();
+	while (total_write < nbytes) {
+		ret = write(fd, buf + total_write, nbytes - total_write);
+		if (ret != nbytes - total_write) {
+			printf("%s errno: %d\n", __FUNCTION__, errno);
+			close(fd);
+			return ERROR;
+		}
+		total_write += ret;
+	}
+	timediff = get_time() - time;
```

Comment:
The get_time works with the gettimeofday. But it works based on the tick. I don't know it's proper resolution of this test. Could you check the unit of this test?

Reply:
The unit of this measurement is millisecond(ms). I updated the unit to logs like `print_log("write:%ldms, ", timediff);`